File: Study Plan/LeetCode 75/1431 Kids With the Greatest Number of Candies/KidsWithTheGreatestNumberOfCandies.cs
/*  
    Problem
    --------------------------------------------------------------------------------
    1431. Kids With the Greatest Number of Candies
*/

/*  
    Solution
    --------------------------------------------------------------------------------
    https://leetcode.com/problems/kids-with-the-greatest-number-of-candies/solutions/4384104/c-linq-time-and-space-complexity-o-n-beats-80-in-runtime/

    - Time and Space Complexity O(n)

    - Runtime 131 ms (Beats 80.21% of users with C#)
    - Memory 45.24 MB (Beats 7.08% of users with C#)
*/

public class Solution
{
    public IList<bool> KidsWithCandies(int[] candies, int extraCandies)
    {
        return candies
            .Select(kid => kid + extraCandies >= candies.Max())
            .ToList();
    }
}

File: Study Plan/LeetCode 75/1431 Kids With the Greatest Number of Candies/KidsWithTheGreatestNumberOfCandies.java
/*  
    Problem
    --------------------------------------------------------------------------------
    1431. Kids With the Greatest Number of Candies
*/

/*  
    Solution
    --------------------------------------------------------------------------------
    - Runtime 4 ms (Beats 6.88% of users with Java)
    - Memory 42.60 MB (Beats 15.27% of users with Java)
*/

class Solution {
    public List<Boolean> kidsWithCandies(int[] candies, int extraCandies) {
        int maxCandies = Arrays.stream(candies).max().orElse(0);
        return Arrays.stream(candies)
            .mapToObj(kid -> kid + extraCandies >= maxCandies)
            .collect(Collectors.toList());
    }
}

